test/specs/DraggingHandlesSpec.js

```diff
@@ -210,6 +219,35 @@ describe("Dragging handles tests", function() {
         // End with mouse up
         testSlider.mouseup(mouseRight);
         expect(testSlider.getValue()).toEqual([2, 5]);
+    });
+
+    it("Should trigger change on mouseup", function() {
+        var ups = 0;
+        var changes = 0;
+
+        testSlider.on('slideStop', function(){
+            expect(changes).toBe(1);
+            ups++;
+        });
+
+        testSlider.mousedown(mouseEvent('mousedown', tickOffsets[1]));
+        expect(testSlider.getValue()).toEqual([1, 5]);
+        expect(ups).toBe(0);
+
+        testSlider.on('change', function(){
+            expect(ups).toBe(0);
+            changes++;
+        });
+
+        testSlider.mouseup(mouseEvent('mouseup', tickOffsets[2]));
+        expect(testSlider.getValue()).toEqual([2, 5]);
+        expect(changes).toBe(1);
+        expect(ups).toBe(1);
+
+        testSlider._mouseup(mouseEvent('mouseup', tickOffsets[1]));
```

Comment:
You can remove this test. When you call `_mouseup()` with a different tick offset, the value should be recalculated and trigger a `change` event.

What's actually happening though, is that after `_mouseup()` is called the first time, `this._state.dragged` is set to `null`. Any calls to `_mouseup()` after that will actually store the new percentage in the `percentage[]` array with the key `'null'`.

```js
// this._state.dragged is 'null'
this._state.percentage['null'] = percentage;
```

https://stackoverflow.com/questions/9687866/is-null-a-valid-javascript-property-name

https://github.com/seiyria/bootstrap-slider/blob/642b40f80790b7c82e9ff149cdf07e40907ccd34/src/js/bootstrap-slider.js#L1733-L1735

https://github.com/seiyria/bootstrap-slider/blob/642b40f80790b7c82e9ff149cdf07e40907ccd34/src/js/bootstrap-slider.js#L1756-L1757

Reply:
Ok. My intend was to make sure, that change is not triggered, if the mouse has not moved.
I will remove the second mouseup.